src/appleseed/renderer/modeling/object/curveobjectwriter.cpp

```diff
@@ -58,23 +60,185 @@ namespace renderer
 
 namespace
 {
-    template <typename CurveType>
-    void write_curve(ostream& output, CurveType& curve)
+    //
+    // Curve object walker.
+    //
+
+    class CurveObjectWalker
+            : public ICurveWalker
     {
-        const size_t control_point_count = curve.get_control_point_count();
+    public:
+        CurveObjectWalker(
+                const CurveObject&   object)
+                : m_object(object),
+                  m_curve_count(0),
+                  m_total_vertex_count(0)
+        {
+            create_parameters();
+        }
+
+        size_t get_basis() const override
+        {
+            switch(m_object.get_basis())
+            {
+                case CurveBasis::Bezier:
+                case CurveBasis::Bspline:
+                case CurveBasis::Catmullrom:
+                    return static_cast<unsigned char>(CurveBasis::Bezier);
+            }
 
-        output << control_point_count;
+            return static_cast<unsigned char>(m_object.get_basis());
+        }
 
-        for (size_t p = 0; p < control_point_count; ++p)
+        string get_basis_string()
         {
-            const GVector3& point = curve.get_control_point(p);
-            const GScalar width = curve.get_width(p);
+            string basis;
+
+            switch(m_object.get_basis())
+            {
+                case CurveBasis::Linear:
+                    basis = "linear";
```

Comment:
Return values directly; get rid of `basis`. Also, what about returning a `const char*`?